packages/beacon-node/src/db/repositories/dataColumnSidecarArchive.ts

```diff
@@ -25,21 +24,21 @@ export class DataColumnSidecarArchiveRepository extends PrefixedRepository<Slot,
   }
 
   encodeKeyRaw(prefix: Slot, id: ColumnIndex): Uint8Array {
-    return Buffer.concat([intToBytes(prefix, 4), intToBytes(id, 4)]);
+    return Buffer.concat([encodeNumberForDbKey(prefix, 4), encodeNumberForDbKey(id, 4)]);
```

Comment:
this involves 3 memory allocations + BigInt involved
we can allocate memory once + use DataView instead
(can do that later)

Reply:
If we write number directly to buffer than we save allocation. The function `intToBytes` allocate a new buffer anyway, so I don't see nay performance impact if we update.

Reply:
we need to add more `"into"` apis, in this case`intIntoBytes()` 
also need to separate `bigIntIntoBytes()` vs `intIntoBytes()`
since this is hot functions I suppose we should do it sooner than later